python/mxnet/model.py

```diff
@@ -261,6 +261,24 @@ def kv_updater(index, grad, weight):
     merged_shape = tuple(merged_shape)
     out_cpu_array = nd.zeros(merged_shape, cpu())
 
+    grad_update_order = []
+    for index, pair in enumerate(zip(arg_blocks, grad_blocks)):
+        arg_list, grad_list = pair
+        if grad_list[0] is not None:
+            grad_size = 1
+            for d in grad_list[0].shape:
```

Comment:
Is it possible not to introduce the gradient update order on this side?

We can make use of the space information on C++ side, More comment on PR